railties/lib/rails/generators/rails/sessions/templates/controllers/sessions_controller.rb

```diff
@@ -1,5 +1,6 @@
 class SessionsController < ApplicationController
-  allow_unauthenticated_access
+  allow_unauthenticated_access(only: %i[ new create ])
+  before_action :resume_session, only: :new
```

Comment:
without this line, `authenticate?` in the `#new` action will always return `nil`

Reply:
Yup!
I've submitted similar PR(https://github.com/rails/rails/pull/52373) for this yesterday, I'm ok with everything being contained in this PR tho, I can close my PR

Reply:
I'd solve this other problem at a more general level, within the same concern

Reply:
The `resume_session` bit was fixed in 1dd37e01937f6711cd15267481df8c0446961472. Can you rebase and update the PR title/description with a summary of the remaining changes?